x/evidence/simulation/decoder.go

```diff
@@ -4,14 +4,20 @@ import (
 	"bytes"
 	"fmt"
 
+	"github.com/cosmos/cosmos-sdk/x/evidence/exported"
+
 	tmkv "github.com/tendermint/tendermint/libs/kv"
 
 	"github.com/cosmos/cosmos-sdk/x/evidence/types"
 )
 
+type EvidenceUnmarshaler interface {
+	UnmarshalEvidence([]byte) (exported.Evidence, error)
+}
+
 // NewDecodeStore returns a decoder function closure that unmarshals the KVPair's
 // Value to the corresponding evidence type.
-func NewDecodeStore(cdc types.Codec) func(kvA, kvB tmkv.Pair) string {
```

Comment:
Why doesn't `types.Codec` suffice here anymore?

Reply:
Because it needs to call `UnmarshalEvidence`

Reply:
Right, and `types.Codec` has said interface method.

Reply:
Sorry, the keeper has the codec but we don't want a `GetCodec` method on the keeper, so I just had the keeper implement `UnmarshalEvidence`. Wasn't sure how to handle this without some messy stuff in the app module and this seemed cleanest

Reply:
Or we could make the keeper implement `types.Codec` and just have `Codec` as an embedded field

Reply:
Ohhh I see, I assumed the actual codec was being provided, not the keeper. This is fine.